hazelcast/src/main/java/com/hazelcast/jet/impl/JetClientInstanceImpl.java

```diff
@@ -141,8 +143,12 @@ public Job newJobProxy(long jobId, UUID lightJobCoordinator) {
         return new ClientJobProxy(client, jobId, lightJobCoordinator);
     }
 
-    public Job newJobProxy(long jobId, boolean isLightJob, @Nonnull Object jobDefinition, @Nonnull JobConfig config) {
-        return new ClientJobProxy(client, jobId, isLightJob, jobDefinition, config);
+    public Job newJobProxy(long jobId,
+                           boolean isLightJob,
+                           @Nonnull Object jobDefinition,
+                           @Nonnull JobConfig config,
+                           @Nullable Subject subject) {
+        return new ClientJobProxy(client, jobId, isLightJob, jobDefinition, config, subject);
```

Comment:
this method does not make sense on client side

Reply:
I had a choice: either I duplicate methods or pass nullable `Subject` to existing methods. I prefer latter.

Reply:
you can also throw UnsupportedOperationException with some reasonable message